SampleApplication.py

```diff
@@ -1,40 +1,77 @@
 import AbstractApplication as Base
-from time import sleep
+from threading import Semaphore
+
 
 class SampleApplication(Base.AbstractApplication):
+    # setup our Application
+    def __init__(self):
+        super().__init__()
+
+        # Semaphores for async execution. They make sure that the action is completed.
+        self.gesture_lock = Semaphore(0)
+        self.language_lock = Semaphore(0)
+        self.text_lock = Semaphore(0)
+        self.eyecolor_lock = Semaphore(0)
+        self.name_lock = Semaphore(0)
+
+        # Attributes of our application
+        self.user_name = None
+
     def main(self):
+        # Setting language
         self.setLanguage('en-US')
+        self.language_lock.acquire()
+
+        # Change Eyecolor
+        self.setEyeColour('red')
+        self.eyecolor_lock.acquire()
```

Comment:
I think we should acquire semaphores before the action we want to take since we want to "lock" the thread so that nothing interferes. Correct me if I'm wrong though :)

Reply:
As far as I got it, as soon as acquire is called, the thread block. So if we acquire it before the action, the program will just stop working. In the sample code it is also done like that. We will have to try it out I think.